health-services/household/src/main/java/org/egov/household/service/HouseholdMemberService.java

```diff
@@ -0,0 +1,281 @@
+        }
+        return columnName;
+    }
+
+    private String getHouseholdId(HouseholdMember householdMember) {
+        return householdMember.getHouseholdId()!=null
+                ? householdMember.getHouseholdId() :
+                householdMember.getHouseholdClientReferenceId();
+    }
+
+    private void validateHeadOfHousehold(HouseholdMember householdMember) {
+        if(householdMember.getIsHeadOfHousehold()){
+            List<HouseholdMember> householdMembersHeadCheck = householdMemberRepository.findIndividualByHousehold(householdMember.getHouseholdId()).stream().filter(
+                    householdMember1 -> householdMember1.getIsHeadOfHousehold())
+                    .collect(Collectors.toList());
+
+            if(!householdMembersHeadCheck.isEmpty()){
+                throw new CustomException("HOUSEHOLD_ALREADY_HAVE_HEAD_OF_HOUSEHOLD", householdMember.getIndividualId());
+            }
+        }
+    }
+
+    private List<HouseholdMember> validateIndividualMapping(HouseholdMember householdMember, IndividualResponse searchResponse) {
+        List<Individual> individuals = searchResponse.getIndividual();
+        if(individuals.isEmpty()){
+            throw new CustomException("INDIVIDUAL_NOT_FOUND", householdMember.getIndividualId());
+        }
+
+        Individual individual = individuals.get(0);
+        householdMember.setIndividualId(individual.getId());
+        householdMember.setIndividualClientReferenceId(individual.getClientReferenceId());
+
+        List<HouseholdMember> individualSearchResult = householdMemberRepository.findIndividual(householdMember.getIndividualId());
+        return individualSearchResult;
+    }
+
+
+    public List<HouseholdMember> search(HouseholdMemberSearch householdMemberSearch, Integer limit, Integer offset, String tenantId,
+                                        Long lastChangedSince, Boolean includeDeleted) {
+
+        String idFieldName = getIdFieldName(householdMemberSearch);
+        if (isSearchByIdOnly(householdMemberSearch, idFieldName)) {
+            List<String> ids = new ArrayList<>();
+            ids.add((String) ReflectionUtils.invokeMethod(getIdMethod(Collections
+                            .singletonList(householdMemberSearch)),
+                    householdMemberSearch));
+            return householdMemberRepository.findById(ids,
+                    idFieldName, includeDeleted).stream()
+                    .filter(lastChangedSince(lastChangedSince))
+                    .filter(havingTenantId(tenantId))
+                    .filter(includeDeleted(includeDeleted))
+                    .collect(Collectors.toList());
+        }
+        try {
+            return householdMemberRepository.find(householdMemberSearch, limit, offset,
+                    tenantId, lastChangedSince, includeDeleted);
+        } catch (QueryBuilderException e) {
+            throw new CustomException("ERROR_IN_QUERY", e.getMessage());
```

Comment:
would this ever happen since the query is being constructed internally?

Reply:
This is generic implementation